apps/app/src/components/OpacityElement.tsx

```diff
@@ -0,0 +1,18 @@
+import { ReactNode } from "react";
+import { twMerge } from "tailwind-merge";
+
+type OpacityElementProps = {
+	children: ReactNode;
+	pernament?: boolean;
+	className?: string;
+};
+
+export const OpacityElement = ({ children, pernament, className }: OpacityElementProps) => {
```

Comment:
Tworzenie komponentu w celu dodawania 1 klasy wydaje się zbyteczne. Czemu po prostu nie użyć tej klasy tam, gdzie jest potrzebna?

Reply:
@mmiszy, zgadam się z tym, że może wydawać się to zbyteczne. Przyznać trzeba jednak, iż łatwiej w ten sposób wprowadzać zmiany, ponieważ wystarczy wprowadzić je w jednym miejscu - w komponencie `OpacityElement`. Dodatkowy powód dla którego postanowiłem stworzyć do tego ten komponent to fakt, iż czasem chcę dodać ten efekt przezroczystości na elementach / komponentach, które nie przyjmują właściwości `className` - np. jest tak w przypadku komponentu `ActiveNavigationLink` wykorzystywanego w `HeaderNavigation`.

Krótko zalety tego rozwiązania:
1. zmiany wprowadzam w jednym miejscu, dzięki temu mniejsze prawdopodobieństwo, że pominę jakiś element,
2. brak konieczności edycji komponentów, tak aby można było do nich dodać `className`.

Czy Twoim zdaniem nadal lepiej jest po prostu dodać ręcznie klasę do każdego elementu, gdzie chcę zastosować przeźroczystość?

Reply:
Tak, lepiej dać klasę. Ten kod to idealny przykład overengineering naprawdę prostej rzeczy.

Reply:
Dzięki za zwrócenie uwagi, dość często mi się to zdarza...